### DISCUSSION TITLE
Relations not created at all by introspect (prisma example)

### DISCUSSION THREAD

**Discussion by TomasMoskala** [prisma/prisma#2701] [Q&A] [answered]
Hi, 
my model does not contain any relation fields.
I tried the example tables for My SQL
https://www.prisma.io/docs/guides/database-workflows/foreign-keys/mysql

I ru thie in phpMyAdmin

```sql
CREATE TABLE `User` (
  `id` INT AUTO_INCREMENT PRIMARY KEY,
  `name` VARCHAR(256)
);
CREATE TABLE `Post` (
  `id` INT AUTO_INCREMENT PRIMARY KEY,
  `title` VARCHAR(256),
  `authorId` INT,
  CONSTRAINT `author` FOREIGN KEY (`authorId`) REFERENCES `User`(`id`)
);
```

Output

```sql
CREATE TABLE `User` ( `id` INT AUTO_INCREMENT PRIMARY KEY, `name` VARCHAR(256) ) 
CREATE TABLE `Post` ( `id` INT AUTO_INCREMENT PRIMARY KEY, `title` VARCHAR(256), `authorId` INT, CONSTRAINT `author` FOREIGN KEY (`authorId`) REFERENCES `User`(`id`) ) 
```

Then I run this in VS Code

```
npx prisma introspect
Environment variables loaded from ./prisma/.env

Introspecting based on datasource defined in prisma\schema.prisma …

✔ Introspected 6 models and wrote them into prisma\schema.prisma in 628ms

Run prisma generate to generate Prisma Client.
```

My models look like this

```prisma
model Post {
  authorId Int?
  id       Int     @default(autoincrement()) @id
  title    String?

  @@index([authorId], name: "author")
}

model User {
  id   Int     @default(autoincrement()) @id
  name String?
}
```

Whereas the example page shows this

```prisma
model Post {
  authorId Int?
  id       Int     @default(autoincrement()) @id
  title    String?
  User     User?   @relation(fields: [authorId], references: [id])
  @@index([authorId], name: "author")
}
model User {
  id   Int     @default(autoincrement()) @id
  name String?
  Post Post[]
}
```

I have no clue what I did wrong, thanks in advance.

**Comment by TomasMoskala** [comment, marked as answer]
WTH I was using ENGINE = MyISAM by default which does not enforce constraints
ENGINE = InnoDB is the correct syntax

Solved myself :)

The examples should  be updated explicitly as there is no mention about this

CREATE TABLE `Post` (
  `id` INT AUTO_INCREMENT PRIMARY KEY,
  `title` VARCHAR(256),
  `authorId` INT,
  CONSTRAINT `author` FOREIGN KEY (`authorId`) REFERENCES `User`(`id`)
) ENGINE = InnoDB;

**Reply by ryands17** [reply]
Hey @TomasMoskala :wave: I think the default engine in MySQL is InnoDB which is why we have not specified it. But we can add that explicitly to the docs as well :)

**Reply by janpio** [reply]
I actually agree - on older MySQL versions the default is not InnoDB so this is a problem for some users. Can you please create a docs issue @ryands17?

**Reply by ryands17** [reply]
@janpio referenced: https://github.com/prisma/docs/issues/432

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
